Error: 'MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE' undeclared, while building with cmake.

### Summary
After cloning the project , when I try to build it using cmake  It throws an error. 

`error.c:108:25: error: ‘MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE’ undeclared`

### System information

> Mbed TLS version (number or commit id): 
- Version : Mbed TLS 3.3.0
- Last commit ```8c89224991adff88d53cd380f42a2baa36f91454```

> Operating system and version: 
- Operating System : Linux Mint 21 Cinnamon
- Cinnamon version : 5.4.12
- Kernel : 5.15.0-56-generic
- Configuration (if not default, please attach `mbedtls_config.h`):  Configuration is [default](https://github.com/Mihir-Raj-Singh/mbedtls/blob/262851df1cc4f090b3e2920cd647d5e22c9bb2f1/include/mbedtls/mbedtls_config.h).

> Compiler and options (if you used a pre-built binary, please indicate how you obtained it): 
- None

> Additional environment information: 
- None

### Expected behavior

- Build process must run successfully and produced the desired files.

### Actual behavior

- Throws an error :

```/home/scorpion/Desktop/Untitled Folder/build/library/error.c: In function ‘mbedtls_strerror’:
/home/scorpion/Desktop/Untitled Folder/build/library/error.c:108:25: error: ‘MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE’ undeclared (first use in this function); did you mean ‘MBEDTLS_SSL_ALL_ALERT_MESSAGES’?
  108 |         if(use_ret == -(MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE))
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         MBEDTLS_SSL_ALL_ALERT_MESSAGES
/home/scorpion/Desktop/Untitled Folder/build/library/error.c:108:25: note: each undeclared identifier is reported only once for each function it appears in
gmake[2]: *** [library/CMakeFiles/mbedcrypto.dir/build.make:551: library/CMakeFiles/mbedcrypto.dir/error.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:813: library/CMakeFiles/mbedcrypto.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
![cmake](https://user-images.githubusercontent.com/92584041/210714678-45127b40-67d6-449a-9886-f5a7d75404fd.png)

```
### Steps to reproduce

- clone the repo on your local system
- `mkdir build`
- `cd build`
- `cmake /path/to/mbedtls/folder/`
- `cmake --build .`


### Additional information

> `cmake /path/to/mbedtls/folder/` output :

![Image 1](https://user-images.githubusercontent.com/92584041/210714855-1a51e523-07d2-4048-994c-7aa1bb616a4f.png)

>  `cmake --build .` output :

![Image 2](https://user-images.githubusercontent.com/92584041/210714887-57d5c6d3-2686-408b-9f54-63c251865e15.png)

Hey @tom-daubney-arm , It's alright let's do some more investigation on this issue. Will be a great opportunity for me to learn as a beginner in programming and in open source. 

I did try cloning the project from scratch , it occurs every time.

- **Here are the cmake outputs -**

```
cmake ~/Desktop/Untitled\ Folder/mbedtls/              

-- The C compiler identification is GNU 9.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Python3: /usr/bin/python3.10 (found version "3.10.6") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- Found Perl: /usr/bin/perl (found version "5.34.0") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/scorpion/Desktop/Untitled Folder/build

```

```
cmake --build .                              
[  0%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/asn1_helpers.c.o
[  0%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/bignum_helpers.c.o
[  0%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/certs.c.o
[  0%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/hash.c.o
[  0%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/platform_builtin_keys.c.o
[  1%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_aead.c.o
[  1%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_asymmetric_encryption.c.o
[  1%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_cipher.c.o
[  1%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_key_agreement.c.o
[  1%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_key_management.c.o
[  1%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_mac.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/drivers/test_driver_signature.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/fake_external_rng_for_test.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/helpers.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/psa_crypto_helpers.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/psa_exercise_key.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/random.c.o
[  2%] Building C object CMakeFiles/mbedtls_test.dir/tests/src/threading_helpers.c.o
[  2%] Built target mbedtls_test
[  2%] Generating version_features.c
[  2%] Generating error.c
readline() on closed filehandle FILE at /home/scorpion/Desktop/Untitled Folder/mbedtls/library/../scripts/generate_errors.pl line 68.
Use of uninitialized value $content in pattern match (m//) at /home/scorpion/Desktop/Untitled Folder/mbedtls/library/../scripts/generate_errors.pl line 71.
[  2%] Generating psa_crypto_driver_wrappers.c
[  2%] Building C object library/CMakeFiles/mbedcrypto.dir/aes.c.o
[  3%] Building C object library/CMakeFiles/mbedcrypto.dir/aesni.c.o
[  3%] Building C object library/CMakeFiles/mbedcrypto.dir/aria.c.o
[  3%] Building C object library/CMakeFiles/mbedcrypto.dir/asn1parse.c.o
[  3%] Building C object library/CMakeFiles/mbedcrypto.dir/asn1write.c.o
[  3%] Building C object library/CMakeFiles/mbedcrypto.dir/base64.c.o
[  3%] Building C object library/CMakeFiles/mbedcrypto.dir/bignum.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/bignum_core.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/bignum_mod.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/bignum_mod_raw.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/camellia.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/ccm.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/chacha20.c.o
[  4%] Building C object library/CMakeFiles/mbedcrypto.dir/chachapoly.c.o
[  5%] Building C object library/CMakeFiles/mbedcrypto.dir/cipher.c.o
[  5%] Building C object library/CMakeFiles/mbedcrypto.dir/cipher_wrap.c.o
[  5%] Building C object library/CMakeFiles/mbedcrypto.dir/constant_time.c.o
[  5%] Building C object library/CMakeFiles/mbedcrypto.dir/cmac.c.o
[  5%] Building C object library/CMakeFiles/mbedcrypto.dir/ctr_drbg.c.o
[  5%] Building C object library/CMakeFiles/mbedcrypto.dir/des.c.o
[  6%] Building C object library/CMakeFiles/mbedcrypto.dir/dhm.c.o
[  6%] Building C object library/CMakeFiles/mbedcrypto.dir/ecdh.c.o
[  6%] Building C object library/CMakeFiles/mbedcrypto.dir/ecdsa.c.o
[  6%] Building C object library/CMakeFiles/mbedcrypto.dir/ecjpake.c.o
[  6%] Building C object library/CMakeFiles/mbedcrypto.dir/ecp.c.o
[  6%] Building C object library/CMakeFiles/mbedcrypto.dir/ecp_curves.c.o
[  7%] Building C object library/CMakeFiles/mbedcrypto.dir/entropy.c.o
[  7%] Building C object library/CMakeFiles/mbedcrypto.dir/entropy_poll.c.o
[  7%] Building C object library/CMakeFiles/mbedcrypto.dir/error.c.o
/home/scorpion/Desktop/Untitled Folder/build/library/error.c: In function ‘mbedtls_strerror’:
/home/scorpion/Desktop/Untitled Folder/build/library/error.c:108:25: error: ‘MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE’ undeclared (first use in this function); did you mean ‘MBEDTLS_SSL_ALL_ALERT_MESSAGES’?
  108 |         if(use_ret == -(MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE))
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         MBEDTLS_SSL_ALL_ALERT_MESSAGES
/home/scorpion/Desktop/Untitled Folder/build/library/error.c:108:25: note: each undeclared identifier is reported only once for each function it appears in
gmake[2]: *** [library/CMakeFiles/mbedcrypto.dir/build.make:551: library/CMakeFiles/mbedcrypto.dir/error.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:813: library/CMakeFiles/mbedcrypto.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2

```







> both the configuration and the actual build output as text,

Can you please be more specific which configuration file you are talking about ? (If you need anything else let me know.)

--- a/ChangeLog.d/fix_build_for_directory_names_containing_spaces.txt
+++ b/ChangeLog.d/fix_build_for_directory_names_containing_spaces.txt
@@ -1,0 +1,4 @@
+Bugfix
+   * Fix a bug in the build where directory names containing spaces were
+     causing generate_errors.pl to error out resulting in a build failure.
+     Fixes issue #6879.

--- a/scripts/generate_errors.pl
+++ b/scripts/generate_errors.pl
@@ -60,11 +60,11 @@ open(FORMAT_FILE, '<:crlf', "$error_format_file") or die "Opening error format f
 my $error_format = <FORMAT_FILE>;
 close(FORMAT_FILE);
 
-my @files = <$include_dir/*.h>;
+my @files = glob qq("$include_dir/*.h");
 my @necessary_include_files;
 my @matches;
 foreach my $file (@files) {
-    open(FILE, '<:crlf', "$file");
+    open(FILE, '<:crlf', $file) or die("$0: $file: $!");
     my $content = <FILE>;
     close FILE;
     my $found = 0;

